Sources/Public/Animation/LottieView.swift

```diff
@@ -115,26 +115,23 @@ public struct LottieView<Placeholder: View>: UIViewConfiguringSwiftUIView {
   // MARK: Public
 
   public var body: some View {
-    ZStack {
-      if let animationSource {
-        LottieAnimationView.swiftUIView {
-          defer { animationDidLoad?(animationSource) }
-          return LottieAnimationView(
-            animationSource: animationSource,
-            imageProvider: imageProviderConfiguration?.imageProvider,
-            textProvider: textProvider,
-            fontProvider: fontProvider,
-            configuration: configuration,
-            logger: logger)
-        }
-        .sizing(sizing)
-        .configure { context in
-          applyCurrentAnimationConfiguration(to: context.view)
-        }
-        .configurations(configurations)
-      } else {
-        placeholder?()
-      }
+    LottieAnimationView.swiftUIView {
+      LottieAnimationView(
+        animationSource: animationSource,
+        imageProvider: imageProviderConfiguration?.imageProvider,
+        textProvider: textProvider,
+        fontProvider: fontProvider,
+        configuration: configuration,
+        logger: logger)
+    }
+    .sizing(sizing)
+    .configure { context in
+      applyCurrentAnimationConfiguration(to: context.view)
+    }
+    .configurations(configurations)
+    .overlay {
```

Comment:
This is a behavior change that will probably affect existing consumers (see comparison below) but I think this is ok because the updated behavior is seems more expected and predictable

```swift
LottieView {
  try await loadAnimation()
} placeholder: {
  LoadingIndicator()
    .frame(width: 50, height: 50)
}
```

| Before | After |
| ---- | ---- |
| <img width="731" alt="Screenshot 2024-01-19 at 4 29 11 PM" src="https://github.com/airbnb/lottie-ios/assets/1811727/0258de34-9b9b-447d-8f92-eb9f1ff4edb2"> | <img width="731" alt="Screenshot 2024-01-19 at 4 31 28 PM" src="https://github.com/airbnb/lottie-ios/assets/1811727/110e5e0c-4827-497f-b6ee-22ec7586513b"> |